src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1550,15 +1550,9 @@ public virtual IndexProgress GetProgress(DocumentsOperationContext documentsCont
             }
         }
 
-        public IndexProgress.CollectionStats GetOrCreateStats(string collection)
+        public IndexProgress.CollectionStats GetStats(string collection)
         {
-            if (_inMemoryIndexProgress.TryGetValue(collection, out var stats) == false)
-            {
-                stats = new IndexProgress.CollectionStats();
-                _inMemoryIndexProgress.TryAdd(collection, stats);
-            }
-
-            return stats;
+            return _inMemoryIndexProgress.GetOrAdd(collection, new IndexProgress.CollectionStats());
```

Comment:
Can we use lambda in add? This will create instance of CollectionStats even if value exists.

Reply:
thanks
fixed